testhelper/testhelper.go

```diff
@@ -139,41 +169,104 @@ func IntegrationLidarTimedSensor(
 	readingTime := time.Date(2021, 8, 15, 14, 30, 45, 100, time.UTC)
 
-	ts.TimedSensorReadingFunc = func(ctx context.Context) (s.TimedSensorReadingResponse, error) {
+	ts.TimedLidarSensorReadingFunc = func(ctx context.Context) (s.TimedLidarSensorReadingResponse, error) {
 		readingTime = readingTime.Add(sensorReadingInterval)
-		t.Logf("TimedSensorReading Mock i: %d, closed: %v, readingTime: %s\n", i, closed, readingTime.String())
+		t.Logf("TimedLidarSensorReading Mock i: %d, closed: %v, readingTime: %s\n", i, closed, readingTime.String())
 		if i >= NumPointClouds {
 			// communicate to the test that all lidar readings have been written
 			if !closed {
 				done <- struct{}{}
 				closed = true
 			}
-			return s.TimedSensorReadingResponse{}, errors.New("end of dataset")
+			return s.TimedLidarSensorReadingResponse{}, errors.New("end of dataset")
 		}
 
 		file, err := os.Open(artifact.MustPath("viam-cartographer/mock_lidar/" + strconv.FormatUint(i, 10) + ".pcd"))
 		if err != nil {
-			t.Error("TEST FAILED TimedSensorReading Mock failed to open pcd file")
-			return s.TimedSensorReadingResponse{}, err
+			t.Error("TEST FAILED TimedLidarSensorReading Mock failed to open pcd file")
+			return s.TimedLidarSensorReadingResponse{}, err
 		}
 		readingPc, err := pointcloud.ReadPCD(file)
 		if err != nil {
-			t.Error("TEST FAILED TimedSensorReading Mock failed to read pcd")
-			return s.TimedSensorReadingResponse{}, err
+			t.Error("TEST FAILED TimedLidarSensorReading Mock failed to read pcd")
+			return s.TimedLidarSensorReadingResponse{}, err
 		}
 
 		buf := new(bytes.Buffer)
 		err = pointcloud.ToPCD(readingPc, buf, pointcloud.PCDBinary)
 		if err != nil {
-			t.Error("TEST FAILED TimedSensorReading Mock failed to parse pcd")
-			return s.TimedSensorReadingResponse{}, err
+			t.Error("TEST FAILED TimedLidarSensorReading Mock failed to parse pcd")
+			return s.TimedLidarSensorReadingResponse{}, err
 		}
 
 		i++
-		return s.TimedSensorReadingResponse{Reading: buf.Bytes(), ReadingTime: readingTime, Replay: replay}, nil
+		return s.TimedLidarSensorReadingResponse{Reading: buf.Bytes(), ReadingTime: readingTime, Replay: replay}, nil
+	}
+
+	return ts, nil
+}
+
+// IntegrationTimedIMUSensor returns a mock timed IMU sensor.
+// When the mock is called, it returns the next mock IMU readings, with the
+// ReadingTime incremented by the sensorReadingInterval.
+// The Replay sensor field of the mock readings will match the replay parameter.
+// When the end of the mock IMU readings is reached, the done channel
+// is written to once so the caller can detect all IMU readings have been emitted
+// from the mock. This is intended to match the same "end of dataset" behavior of a
+// replay sensor.
+// It is important to provide deterministic time information to cartographer to
+// ensure test outputs of cartographer are deterministic.
+// Note that IMU replay sensors are not yet fully supported.
+func IntegrationTimedIMUSensor(
```

Comment:
[q] Why arent we using artifacts here?

Reply:
Excuse my ignorance but what are artifacts?

Reply:
Artifacts are what we call the stored files such as PCDs that get pulled from the cloud for testing. You mentioned in a previous comment Nick had suggested doing this inline so im resolving